### DISCUSSION TITLE
No-go for install on new, virgin ER-X

### DISCUSSION THREAD

**Discussion by ghost** [Ysurac/openmptcprouter#1942] [General]
The docs show a procedure for installing OMR on the Ubiquiti Edgerouter-X. I tried this on a new ER-X and was stopped at "add system image". Here's the dialog:

***************

/tmp >> scp openwrt-ramips-mt7621-ubnt-erx-initramfs-factory.tar ubnt@192.168.1.19:/tmp/         
The authenticity of host '192.168.1.19 (192.168.1.19)' can't be established.
ECDSA key fingerprint is SHA256:2Y7raF0HmLJ0Zk5OhXPXM3oPyR4kJeKI/nDBXiSE6qA.
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
Warning: Permanently added '192.168.1.19' (ECDSA) to the list of known hosts.
Welcome to EdgeOS

By logging in, accessing, or using the Ubiquiti product, you
acknowledge that you have read and understood the Ubiquiti
License Agreement (available in the Web UI at, by default,
http://192.168.1.1) and agree to be bound by its terms.

ubnt@192.168.1.19's password: 
openwrt-ramips-mt7621-ubnt-erx-initramfs-factory.tar                                                             100%  118KB   2.2MB/s   00:00    
/tmp >> ssh ubnt@192.168.1.19
Welcome to EdgeOS

By logging in, accessing, or using the Ubiquiti product, you
acknowledge that you have read and understood the Ubiquiti
License Agreement (available in the Web UI at, by default,
http://192.168.1.1) and agree to be bound by its terms.

ubnt@192.168.1.19's password: 
Linux ubnt 3.10.107-UBNT #1 SMP Fri Feb 21 10:42:32 UTC 2020 mips
Welcome to EdgeOS
ubnt@ubnt:~$ add system image /tmp/openwrt-ramips-mt7621-ubnt-erx-initramfs-factory.tar
Checking upgrade image...Upgrade image does not support the device. Upgrade failed.
/home/ubnt
ubnt@ubnt:~$ show system image
The system currently has the following image(s) installed:

v1.10.11.5274269.200221.1028   (running image) (default boot)

**Comment by Ysurac** [comment]
I used this doc: https://openwrt.org/toh/ubiquiti/edgerouter_x_er-x_ka (same as openmptcprouter doc)
Worked on my device

**Reply by ghost** [reply]
Thanks for the link. It pointed to a different image for openwrt 18.06 that the ER-X seemed to find acceptable. I also (mostly) used their backup suggestions, then followed the "Router install" guidance for OMR. After booting OMR I moved the computer from eth1 to eth0 on the ER-X and changed the IP to OMR's default 192.168.100.1. It takes some patience to go through the steps...

**Reply by Ysurac** [reply]
Thanks I updated the doc.
Then if you can tell me if OpenMPTCProuter image is working correctly or not :)

**Reply by ghost** [reply]
That image worked for me, but it gives different output to "show system image". Once it is installed and running, you can only get SSH access on eth1. After installing OMR then one needs to go back to eth0.

I am giving up on the ER-X port since performance is terrible and I don't know how to fix it. With two DSL ( 12.5 / 1 each ) and a 4G modem ( 15/15 ) aggregated, I only see something like 13 /3 throughput. Running the same version of OMR on Intel x64 ( Protectli FW4B ) I see throughput around 35 / 15.

**Comment by Adorfer** [comment]
for some devices (ER-X is in my oppinion one of them) the manufactuer likes during production time to make
- changes in the booloader
- changes in the partition layout
- changes in the webUI-updater (accepted filenames, checksum-methods, 'only higher version'-check...)

In other words: 
if installation of an OpenWRT-fork (like OMR) fails:  
first install a standard OpenWRT, then the target "Distro" via _sysupgrade_-image.

**Reply by ghost** [reply]
Thinking about it that way is helpful, but it still takes about the same number of steps. My post was intended to clarify some points for the next person going this route, and hopefully not be viewed as a complaint. It would be fantastic to have an ER-X hosting the local side.
